src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1685,12 +1686,50 @@ fastdds::TopicAttributes DataWriterImpl::get_topic_attributes(
     topic_att.topicName = topic.get_name();
     topic_att.topicDataType = topic.get_type_name();
     topic_att.topicKind = type->m_isGetKeyDefined ? WITH_KEY : NO_KEY;
-    if (type->auto_fill_type_information() && xtypes::TK_NONE != type->type_identifiers().type_identifier1()._d())
+
+    using utils::to_type_propagation;
+    using utils::TypePropagation;
+
+    auto properties = publisher_->get_participant()->get_qos().properties();
+    auto type_propagation = to_type_propagation(properties);
+    bool should_assign_type_information =
+            (TypePropagation::TYPEPROPAGATION_ENABLED == type_propagation) ||
+            (TypePropagation::TYPEPROPAGATION_MINIMAL_BANDWIDTH == type_propagation);
+
+    if (should_assign_type_information && (xtypes::TK_NONE != type->type_identifiers().type_identifier1()._d()))
     {
+        xtypes::TypeInformation type_info;
+
         if (RETCODE_OK ==
                 fastdds::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().get_type_information(
-                    type->type_identifiers(), topic_att.type_information.type_information))
+                    type->type_identifiers(), type_info))
         {
+            switch (type_propagation)
+            {
+                case TypePropagation::TYPEPROPAGATION_ENABLED:
+                {
+                    // Use both complete and minimal type information
+                    topic_att.type_information.type_information = type_info;
+                    break;
+                }
+                case TypePropagation::TYPEPROPAGATION_MINIMAL_BANDWIDTH:
+                {
+                    // Use minimal type information only
+                    topic_att.type_information.type_information.minimal() = type_info.minimal();
+                    break;
+                }
+                case TypePropagation::TYPEPROPAGATION_REGISTRATION_ONLY:
+                {
+                    // Use both complete and minimal type information
+                    topic_att.type_information.type_information = type_info;
+                    break;
+                }
```

Comment:
The conditional on `should_assign_type_information` makes this case impossible to reach

```suggestion
```